test/runtests.jl

```diff
@@ -18,7 +18,6 @@ for meth in filter(f -> isa(eval(f), Function), names(NLPModels))
   @test_throws(NotImplementedError, meth(model))
 end
 
-include("genrose.jl")
```

Comment:
It's not clear to me what lines 12-19 are testing. `model` is defined as a `SimpleNLPModel`, so `obj`, etc., **are** defined and implemented, but the test just isn't using the appropriate calling sequence. `obj(model, x)` works. `obj(model)` doesn't exist but is made to throw `NotImplementedError` because `obj` is declared by default as `obj(nlp :: AbstractNLPModel, args...; kwargs...)`.

To me that's misleading because `obj` should always be `obj(nlp, x)` and never anything else.

I propose to redefine `obj()` as `obj(nlp :: AbstractNLPModel, x :: AbstractVector)`, and similarly for the other methods. Then the test could consist in something like
```
type DummyNLPModel <: AbstractNLPModel
end

dummy = DummyNLPModel()
@test_throws(NotImplementedError, obj(dummy, ones(2))
```
What do you think?